core-java-modules/core-java-collections-maps-7/src/main/java/com/baeldung/map/HashMapWithMaxSizeLimit.java

```diff
@@ -0,0 +1,25 @@
+package com.baeldung.map;
+
+import java.util.HashMap;
+
+public class HashMapWithMaxSizeLimit<K, V> extends HashMap<K, V> {
+
+    private static final long serialVersionUID = 1L;
+
+    private int maxSize = -1;
+    
+   
+    public HashMapWithMaxSizeLimit(int maxSize) {
```

Comment:
Should values smaller than -1 and 0 be allowed here?

Reply:
No, the -1 is used to define an implicit "no limit" default. I forgot to write some explanation in the article, I did it now.